lib/routes/account.js

```diff
@@ -470,19 +470,22 @@ module.exports = function (
         function extractUnblockCode (e) {
           // If it's a customs-related error...
           if (e.errno === error.ERRNO.REQUEST_BLOCKED || e.errno === error.ERRNO.THROTTLED) {
-            request.emitMetricsEvent('account.login.blocked')
-            var verificationMethod = e.output.payload.verificationMethod
-            // ...and if they can verify their way around it,
-            // then extract any unblockCode from the request.
-            // We'll re-throw the error later if the code is invalid.
-            if (verificationMethod === 'email-captcha') {
-              unblockCode = request.payload.unblockCode
-              if (unblockCode) {
-                unblockCode = unblockCode.toUpperCase()
-              }
-              customsErr = e
-              return
-            }
+            return request.emitMetricsEvent('account.login.blocked')
```

Comment:
There were a couple of places, here and another a bit further down, where I needed to propagate the flow event promise in order to make unit tests pass. It means the response time increases a little bit but the alternative was to either lose test coverage or rely on an ugly `setImmediate` (or whatever) hack in the tests.